### PR TITLE
id-validation

### PR DESCRIPTION
**Pull request by Mahmoudkamel50**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -42,6 +42,13 @@
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<!-- https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-starter-validation -->
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-validation</artifactId>
+			<version>2.7.1</version>
+		</dependency>
+
```
Comment (abdelghanycu) [inline pom.xml:51]: what is the need of this dependency?

File: src/main/java/com/example/test/controllers/EmployeeController.java [hunk 2]
```diff
@@ -9,23 +9,28 @@
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+
+
 @RestController
 @RequestMapping("/employee")
 @RequiredArgsConstructor
+
```
Comment (abdelghanycu) [inline src/main/java/com/example/test/controllers/EmployeeController.java]: why did you add this empty lines?

File: src/main/java/com/example/test/services/EmployeeService.java [hunk 3]
```diff
@@ -15,12 +15,12 @@ public class EmployeeService {
 
     @Autowired
     private EmployeeRepo employeeRepo;
-
     public List<Employee> getEmployees() {
         return ( List<Employee>)employeeRepo.findAll();
     }
 
     public Employee getEmployee(Long empId) {
+
```
Comment (abdelghanycu) [inline src/main/java/com/example/test/services/EmployeeService.java]: why did you add this empty line?

File: src/main/java/com/example/test/controllers/EmployeeController.java [hunk 4]
```diff
@@ -9,23 +9,28 @@
...
+            return new ResponseEntity<>("enter valid number", HttpStatus.BAD_REQUEST);
+        }
         return new ResponseEntity<>(employeeService.getEmployee(employeeId), HttpStatus.OK);
     }
 
     @GetMapping
     public ResponseEntity<?> getEmployees() {
 
-        return new ResponseEntity<>(employeeService.getEmployees(), HttpStatus.OK);
+        return new ResponseEntity<>( employeeService.getEmployees(), HttpStatus.OK);
```
Comment (abdelghanycu) [inline src/main/java/com/example/test/controllers/EmployeeController.java]: why you added the space?

File: src/main/java/com/example/test/controllers/EmployeeController.java [hunk 5]
```diff
@@ -9,23 +9,28 @@
...
     @Autowired
     private EmployeeService employeeService;
 
     @GetMapping(path ="{employeeId}")
-    public ResponseEntity<?> getEmployee(@PathVariable Long employeeId) {
-
+    public ResponseEntity<?> getEmployee(@PathVariable Long employeeId){
+        if (employeeId < 1){
+            return new ResponseEntity<>("enter valid number", HttpStatus.BAD_REQUEST);
+        }
```
Comment (abdelghanycu) [inline src/main/java/com/example/test/controllers/EmployeeController.java]: Can you try to use the annotation @Min
https://spring.io/guides/gs/validating-form-input/

File: pom.xml (hunk 1 above)
Comment (Mahmoudkamel50) [reply pom.xml:51]: I tried to use the validation way with @Min.

File: src/main/java/com/example/test/controllers/EmployeeController.java (hunk 2 above)
Comment (Mahmoudkamel50) [reply src/main/java/com/example/test/controllers/EmployeeController.java]: Resolved

File: src/main/java/com/example/test/services/EmployeeService.java (hunk 3 above)
Comment (Mahmoudkamel50) [reply src/main/java/com/example/test/services/EmployeeService.java]: Resolved

File: src/main/java/com/example/test/exceptions/HandleException.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class HandleException {
+    @ExceptionHandler (MethodArgumentNotValidException.class)
+    public ResponseEntity<?>handleCityNotFoundException(MethodArgumentNotValidException ex){
+        return new ResponseEntity<>(ex.getMessage(), HttpStatus.BAD_REQUEST);
+    }
+}
```
Comment (abdelghanycu) [inline src/main/java/com/example/test/exceptions/HandleException.java]: What is the need of this?

File: src/main/java/com/example/test/models/Employee.java [hunk 7]
```diff
@@ -6,6 +6,7 @@
 import lombok.NoArgsConstructor;
 
 import javax.persistence.*;
+import javax.validation.constraints.Min;
```
Comment (abdelghanycu) [inline src/main/java/com/example/test/models/Employee.java]: No need for this import

File: src/main/java/com/example/test/services/EmployeeService.java [hunk 8]
```diff
@@ -15,14 +13,11 @@ public class EmployeeService {
...
     private EmployeeRepo employeeRepo;
-
     public List<Employee> getEmployees() {
         return ( List<Employee>)employeeRepo.findAll();
     }
 
-    public Employee getEmployee(Long empId) {
-        return employeeRepo.findByEmployeeId(empId);
-    }
+    public Employee getEmployee(Long empId) {return employeeRepo.findByEmployeeId(empId);}
```
Comment (abdelghanycu) [inline src/main/java/com/example/test/services/EmployeeService.java]: PLEASE ROLLBACK

**Review by abdelghanycu — APPROVED** [review]
